backend/src/models/email_confirmation.rs

```diff
@@ -0,0 +1,45 @@
+use crate::entities::{email_confirmation, users};
+use crate::models::user::UserInformation;
+use crate::routes::auth::AuthError::{ConfirmationCodeExpired, InvalidConfirmationCode};
+use crate::utils::error::Error;
+use sea_orm::ActiveValue::Set;
+use sea_orm::{ActiveModelTrait, DatabaseConnection, EntityTrait, TransactionTrait};
+
+impl email_confirmation::Model {
+    pub async fn confirm_email(
+        database: &DatabaseConnection,
+        confirmation_code: String,
```

Comment:
Models are responsible for bare logic related to email, make it use &self, and convert things confirmation code to entity  somewhere else

Reply:
Or even better:

email_confirmation:
- get from confirmation_code (make it &str)
- remove

user:
- update `create_with_password` method

auth service:
- combine entire logic (email::get_from_confirm_code, create_with_password, remove)